arkouda/dataframe.py

```diff
@@ -2111,6 +2152,33 @@ def coargsort(self, keys, ascending=True):
             i = i[arange(self.size - 1, -1, -1)]
         return i
 
+    def __reindex(self, idx):
+        if isinstance(self.index, MultiIndex):
+            new_index = MultiIndex(self.index[idx].values, name=self.index.name, names=self.index.names)
+        elif isinstance(self.index, Index) and hasattr(self.index, "name"):
+            new_index = Index(self.index[idx], name=self.index.name)
+        else:
+            new_index = Index(self.index[idx])
```

Comment:
I don't think the `hasattr(self.index, "name")` is doing anything here since an `Index` will always have a name attribute even if it is set to `None`